### ISSUE TITLE
Make tf.image.resize gradient computation compatible with XLA compilation

### ISSUE THREAD

**Issue by prouast** [tensorflow/tensorflow#57575]
### Issue Type

Feature Request

### Source

binary

### Tensorflow Version

tf 2.11.0-dev20220831

### Custom Code

No

### Current Behaviour?

```shell
Currently, XLA compilation does not work for gradient computation of `tf.image.resize`, for example for the default method `ResizeMethod.BILINEAR`:

> ResizeBilinearGrad with align_corners=False or half_pixel_centers=True is not yet implemented

Support for forward pass has recently been added: https://github.com/tensorflow/tensorflow/issues/46447

This makes it impossible to XLA compile models for training that use resizing, such as RetinaNet.
```

### Standalone code to reproduce the issue

```shell
Colab to reproduce error: https://colab.research.google.com/drive/1eevpjtJYqQ7Mkg7PujJu_I5UOfL1PnWX?usp=sharing
```

**Comment by sushreebarsa** [comment]
@sachinprasadhs I was able to replicate the issue on colab, please find the [gist](https://colab.research.google.com/gist/sushreebarsa/c178e2bf561a1a1b786fb3dce47480d9/untitled123.ipynb) here.
Thank you!

**Comment by shkarupa-alex** [comment]
Still waiting this.
A lot of semantic segmentation models use bilinear upsampling. Without gradient xla support we can't use jit_compile=True in model.compile(). I know about autoclustering, but in my experience it works worse then jit_compile (uses more memory -> requires smaller batch size)

**Comment by paynecl** [comment]
Thanks for flagging this again.  Let me assess how we would
prioritize/schedule this and get back to you.

**Comment by paynecl** [comment]
We are working on a fix for this.  Delivery should be expected in early
July.

**Comment by sachinprasadhs** [comment]
Hi @prouast ,

This issue seems to be fixed by https://github.com/tensorflow/tensorflow/commit/58308810880c6746cfa726d1108ff8f9c14a2248 and here is the working [Gist](https://gist.github.com/sachinprasadhs/27f0939147799d1886f3f889e68a0ee8) in Tf-Nightly.
Please verify and feel free to close the issue. Thanks!

**Comment by shkarupa-alex** [comment]
@sachinprasadhs , bilinear interpolation grad now (tf 2.13.0) works in XLA.
But non nearest one (which has forward XLA implementation).

```python
2023-07-06 00:31:29.455196: W tensorflow/core/framework/op_kernel.cc:1828] OP_REQUIRES failed at xla_ops.cc:503 : INVALID_ARGUMENT: Detected unsupported operations when trying to compile graph __inference_train_815[_XlaMustCompile=true,config_proto=10632308226793381189,executor_type=11160318154034397263] on XLA_GPU_JIT: ResizeNearestNeighborGrad (No registered 'ResizeNearestNeighborGrad' OpKernel for XLA_GPU_JIT devices compatible with node {{node gradient_tape/model_1/tf.image.resize_1/resize/ResizeNearestNeighborGrad}}){{node gradient_tape/model_1/tf.image.resize_1/resize/ResizeNearestNeighborGrad}}
```

You can reproduce error with code from colab link in first message. Just add `method=tf.image.ResizeMethod.NEAREST_NEIGHBOR` to `tf.image.resize`

**Comment by sachinprasadhs** [comment]
@changm, Could you please check this issue, which is related to your recent commit. Thanks!

**Comment by changm** [comment]
Do you mean @mikedelorimier as [commit](https://github.com/tensorflow/tensorflow/commit/58308810880c6746cfa726d1108ff8f9c14a2248) is the author? Or did you have a different commit I wrote @sachinprasadhs ?

**Comment by sachinprasadhs** [comment]
> Do you mean @mikedelorimier as [commit](https://github.com/tensorflow/tensorflow/commit/58308810880c6746cfa726d1108ff8f9c14a2248) is the author? Or did you have a different commit I wrote @sachinprasadhs ?

Sorry my bad, that is the right commit and it was from @mikedelorimier . Thanks for correcting me.

**Comment by milmor** [comment]
layers.UpSampling2D(2,  interpolation='nearest') does not work with XLA in tf 2.12 :( . Has anyone tried with tf 2.13?? it would be so great to have this operation, specifically for generative models, This is a bottleneck in Tensorflow, it's so sad that we can't leverage XLA just because of this layer.

**Comment by kmkolasinski** [comment]
@milmor  did you try keras.layers.Resizing ?  it seems it works with XLA. Transpose conv also works

**Comment by milmor** [comment]
@kmkolasinski  layers.Resizing(64, 64, interpolation="nearest") not working with tf 2.13 :( so sad. Unfortunately with bilinear interpolation or transpose, it is not possible to build and replicate a lot of models.

Detected unsupported operations when trying to compile graph __inference_train_step_49273[_XlaMustCompile=true,config_proto=6001324581131673121,executor_type=11160318154034397263] on XLA_GPU_JIT: ResizeNearestNeighborGrad (No registered 'ResizeNearestNeighborGrad' OpKernel for XLA_GPU_JIT

**Comment by kmkolasinski** [comment]
@milmor  the problem is with the "nearest" interpolation, cannot you use "linear" ? which works

**Comment by milmor** [comment]
@kmkolasinski No :( there are some results and behaviors that can only be achieved with nearest interpolation.

**Comment by shkarupa-alex** [comment]
Any chance to fix this finally?

**Comment by Kayyuri** [comment]
Hi @prouast,
Thank you so much for filing this Feature request! We truly appreciate your support. At the moment, this request is not prioritized and mark it for community contributions. If you would like to contribute on a fix, please feel free to fork the repository and submit a pull request in accordance with the TensorFlow [CONTRIBUTION GUIDELINES](https://www.tensorflow.org/community/contribute/code) and link this issue in your PR to ensure the maintainers can easily review and collaborate.
To avoid duplicate work, please leave a comment here if you decide to take it on!

Thank you again for your interest and we look forward to seeing your contributions!
